app/usecase/event/dispatcher.go

```diff
@@ -0,0 +1,64 @@
+package event
+
+import (
+	"errors"
+	"github.com/asaskevich/EventBus"
+	"sync/atomic"
+)
+
+var _ Dispatcher = (*EventDispatcher)(nil)
+var _ Subscriber = (*EventDispatcher)(nil)
+
+// ErrDispatcherIsClosed tells that there is no way to perform manipulations with event dispatcher
+var ErrDispatcherIsClosed = errors.New("failed to perform the operation, the dispatcher is closed")
+
+// EventDispatcher represents the event dispatcher implementation
+type EventDispatcher struct {
+	bus      EventBus.Bus
+	isClosed int32
+}
+
+func (d *EventDispatcher) Dispatch(event Event) error {
+	if atomic.LoadInt32(&(d.isClosed)) != 0 {
```

Comment:
This is my first time seeing `LoadInt32`! What does it do? What does the `0` mean here ( I learnt its meaning only by looking at other pieces of the code)? You should probably extract out `0` and `1` into constants to improve readability.

Reply:
0 - means false here (`isClosed` equals to false)
1 - otherwise

It is a way how to atomically change the variable without using `sync.Lock`

Reply:
@alldroll Please add comments or give the numbers meaningful variable names.